cli/install/install.go

```diff
@@ -0,0 +1,585 @@
+	// installName is a string containing install name of packet.
+	installName string
+}
+
+// DistroInfo is a struct containing info about linux distro.
+type DistroInfo struct {
+	Name         string
+	Vendor       string
+	Version      string
+	Architecture string
+}
+
+var (
+	// PrettyName is a regexp for PrettyName in os-release file.
+	PrettyName = regexp.MustCompile(`^PRETTY_NAME=(.*)$`)
+	// ID is a regexp for ID in os-release file.
+	ID = regexp.MustCompile(`^ID=(.*)$`)
+	// VersionID is a regexp for VersionID in os-release file.
+	VersionID = regexp.MustCompile(`^VERSION_ID=(.*)$`)
+	// TarantoolLink is a link to the git repository of tarantool.
+	TarantoolLink = "https://github.com/tarantool/tarantool.git"
+	// TTLink is a link to the git repository of tt.
+	TTLink = "https://github.com/tarantool/tt"
+)
+
+// getArchitecture returns Architecture of machine.
+func getArchitecture() (string, error) {
+	cmd := exec.Command("uname", "-m")
+	readPipe, writePipe, _ := os.Pipe()
+	cmd.Stdout = writePipe
+	cmd.Stderr = os.Stderr
+	cmd.Start()
+	err := cmd.Wait()
+	writePipe.Close()
+	if err != nil {
+		return "", err
+	}
+
+	var buf bytes.Buffer
+	io.Copy(&buf, readPipe)
+	return buf.String(), err
+}
+
+// getDistroInfo collects info about linux distro.
+func getDistroInfo() DistroInfo {
+	var distroInfo DistroInfo
+	var err error
+
+	distroInfo.Architecture, err = getArchitecture()
+	if err != nil {
+		return distroInfo
+	}
+
+	distoFile, err := os.Open("/etc/os-release")
+	if err != nil {
+		return distroInfo
+	}
+	defer distoFile.Close()
+
+	scanner := bufio.NewScanner(distoFile)
+	for scanner.Scan() {
+		if m := PrettyName.FindStringSubmatch(scanner.Text()); m != nil {
+			distroInfo.Name = strings.Trim(m[1], `"`)
+		} else if m := ID.FindStringSubmatch(scanner.Text()); m != nil {
+			distroInfo.Vendor = strings.Trim(m[1], `"`)
+		} else if m := VersionID.FindStringSubmatch(scanner.Text()); m != nil {
+			distroInfo.Version = strings.Trim(m[1], `"`)
+		}
+	}
+	return distroInfo
+}
+
+// detectOsName returns name of the OS.
+func detectOsName() string {
+	if runtime.GOOS == "darwin" {
+		return "darwin"
+	}
+	if runtime.GOOS == "windows" {
+		return "windows"
+	}
+	if runtime.GOOS == "linux" {
+		distroInfo := getDistroInfo()
+		return distroInfo.Name
+	}
+	return ""
+}
+
+// getLatestTarantoolTag returns latest tag from tarantool repository.
+func getLatestTarantoolTag() (string, error) {
+	cmd := exec.Command("git", "ls-remote", "--tags", "--sort="+
+		"v:refname", TarantoolLink)
+	readPipe, writePipe, _ := os.Pipe()
+	cmd.Stdout = writePipe
+	cmd.Stderr = os.Stderr
+	cmd.Start()
+	err := cmd.Wait()
+	writePipe.Close()
+	if err != nil {
+		return "", err
+	}
+
+	var buf bytes.Buffer
+	io.Copy(&buf, readPipe)
+	versions := buf.String()
+	versionsArray := strings.Split(versions, "\n")
+	if len(versionsArray) == 0 {
+		return "", fmt.Errorf("Could not get latest version of tarantool.")
+	}
+	latestTag := versionsArray[len(versionsArray)-2]
```

Comment:
Leave comment here:
// TODO Use version functions from util/version (will be implemented later)

Reply:
added